### PR TITLE
Add scaleway provider

### PR DESCRIPTION
**Pull request by vanonox**
hi,
this PR is to add Scaleway provider to the DNS API
i have read the guide: https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide
CI is ok
i have edited the wiki page

thanks

### CODE REVIEW

File: dnsapi/dns_scaleway.sh [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  if ! _get_root "$fulldomain"; then
+    _err "invalid domain"
+    return 1
+  fi
+
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  _info "Deleting record"
+  _scaleway_create_TXT_record "$_domain" "$_sub_domain" "$txtvalue"
```
Comment (Neilpang) [inline dnsapi/dns_scaleway.sh:55]: Is this a mistake?
do you mean to use `_scaleway_delete_TXT_record()` here ?

File: dnsapi/dns_scaleway.sh [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  if ! _get_root "$fulldomain"; then
+    _err "invalid domain"
+    return 1
+  fi
+
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  _info "Adding record"
+  _scaleway_create_TXT_record "$_domain" "$_sub_domain" "$txtvalue"
```
Comment (Neilpang) [inline dnsapi/dns_scaleway.sh:31]: why not check the return value of `_scaleway_create_TXT_record()`

File: dnsapi/dns_scaleway.sh [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  return 0
+}
+
+#_acme-challenge.www.domain.com
+#returns
+# _sub_domain=_acme-challenge.www
+# _domain=domain.com
+_get_root() {
+  domain=$1
+  i=2
```
Comment (Neilpang) [inline dnsapi/dns_scaleway.sh:86]: the value of `i` must start from `1`

**Comment by Neilpang** [comment]
see this requirement:
https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide#11-please-read-and-follow-the-instruction-before-creating-a-pull-request

File: dnsapi/dns_scaleway.sh (hunk 1 above)
Comment (vanonox) [reply dnsapi/dns_scaleway.sh:55]: hi , yes , forgot to push this correction

File: dnsapi/dns_scaleway.sh (hunk 2 above)
Comment (vanonox) [reply dnsapi/dns_scaleway.sh:31]: corrected it below

File: dnsapi/dns_scaleway.sh (hunk 3 above)
Comment (vanonox) [reply dnsapi/dns_scaleway.sh]: done thx

**Comment by vanonox** [comment]
hi , i have made the changes
can you review it again ? thanks

**Comment by Neilpang** [comment]
10-please-create-a-new-issue-for-future-bugs

**Comment by vanonox** [comment]
it is the https://github.com/acmesh-official/acme.sh/issues/3295
i have corrected the wiki page (i have put it in the wrong paragraph :/)

**Comment by vanonox** [comment]
no in fact it was good in https://github.com/acmesh-official/acme.sh/wiki/dnsapi/c4b1c9e449636a6522e756397b16ef9b24502805
but moved by someone else after :/

**Pull request merged**